### PR TITLE
Switch to caasp-kvm environment for CI

### PR DESCRIPTION
**Pull request by kiall**
Some pieces stolen from https://github.com/kubic-project/jenkins-library/pull/32 - however caasp-kvm has changed a lot since this PR was opened.

Depends-On: https://github.com/kubic-project/jenkins-library/pull/57

### CODE REVIEW

File: vars/bootstrapEnvironment.groovy [hunk 1]
```diff
@@ -17,21 +17,26 @@ import com.suse.kubic.Minion
...
     timeout(90) {
         try {
             dir('automation/velum-bootstrap') {
                 withEnv([
                     "VERBOSE=true",
-                    "ENVIRONMENT=${WORKSPACE}/terraform/environment.json",
-                    // TODO: drop this after switching to a VM based admin setup
-                    "DEVENV=true",
+                    "ENVIRONMENT=${WORKSPACE}/environment.json",
+                    "DEVENV=${devenv}",
```
Comment (MaximilianMeister) [inline vars/bootstrapEnvironment.groovy]: does it make sense once the caasp-kvm setup is the default to still support the caasp-devenv? i would tend to completely drop it as caasp-kvm will be the new devenv

File: vars/createEnvironmentCaaspKvm.groovy [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            environment.sshKey = environmentJson.sshKey
+
+            environmentJson.minions.each { tfMinion ->
+                Minion minion = new Minion()
+
+                minion.index = tfMinion.index
+                minion.fqdn = tfMinion.fqdn
+                minion.role = tfMinion.role
+                minion.minionId = tfMinion.minionId
+                minion.proxyCommand = tfMinion.proxyCommand != null ? tfMinion.proxyCommand : null
```
Comment (MaximilianMeister) [inline vars/createEnvironmentCaaspKvm.groovy]: same question as in https://github.com/kubic-project/jenkins-library/pull/57/files#r132879961

File: vars/cloneAllKubicRepos.groovy [hunk 3]
```diff
@@ -36,7 +36,7 @@ def call(Map parameters = [:]) {
             cloneKubicRepo(gitBase: gitBase, branch: branch, credentialsId: credentialsId, repo: "terraform")
         },
         'automation': {
-            cloneKubicRepo(gitBase: gitBase, branch: branch, credentialsId: credentialsId, repo: "automation")
+            cloneKubicRepo(gitBase: gitBase, branch: 'heat-environment', credentialsId: credentialsId, repo: "automation")
```
Comment (kiall) [inline vars/cloneAllKubicRepos.groovy]: TODO: Remove this hack

File: vars/createEnvironmentCaaspKvm.groovy (hunk 2 above)
Comment (kiall) [inline vars/createEnvironmentCaaspKvm.groovy]: Done

File: vars/gatherKubicLogs.groovy [hunk 4]
```diff
@@ -16,6 +16,7 @@ import com.suse.kubic.Minion
 
 def call(Map parameters = [:]) {
     Environment environment = parameters.get('environment')
+    boolean realAdmin = parameters.get('realAdmin', false)
```
Comment (grahamhayes) [inline vars/gatherKubicLogs.groovy]: Do we need this now? Are we not just deprecating the caasp-devenv ?

File: vars/withKubicEnvironment.groovy [hunk 5]
```diff
@@ -70,7 +70,11 @@ def call(Map parameters = [:], Closure body) {
             // Gather logs from the environment
             stage('Gather Logs') {
                 try {
-                    gatherKubicLogs(environment: environment)
+                    if (environmentType == 'devenv') {
```
Comment (grahamhayes) [inline vars/withKubicEnvironment.groovy]: As above

File: vars/gatherKubicLogs.groovy (hunk 4 above)
Comment (kiall) [inline vars/gatherKubicLogs.groovy]: Probably not - Once it all works etc, I'll remove these conditionals

**Pull request merged**